docs/kb/rules/using-semgrep-rule-schema-in-vscode.md

```diff
@@ -0,0 +1,71 @@
+---
+description: Use the Semgrep rule schema in VS Code to help make rule writing easier.
+tags:
+  - Rules
+  - VS Code
+---
+
+import MoreHelp from "/src/components/MoreHelp"
+
+# Use the Semgrep rule schema to write rules in VS Code
+
+You may already be familiar with writing rules in the [Semgrep Playground](/docs/playground). However, if your IDE of choice is VS Code and you'd like to write Semgrep rules there, utilizing the Semgrep rule schema can be a great aid while doing so. The Semgrep rule schema will provide a richer editing environment, allowing for VS Code to understand the shape of your rule's YAML file, including its value sets, defaults, and descriptions ([reference](https://marketplace.visualstudio.com/items?itemName=redhat.vscode-yaml#associating-schemas)).
+
+:::tip
+Writing rules locally in your IDE is also helpful when iteratively testing your rule against an entire local repository, as opposed to just a snippet of test code.
+:::
+
+When setup properly, you can expect to get auto-completion options in your VS Code IDE just as you would in the Semgrep Playground when writing rules:
+
+![Example Semgrep YAML rule file with auto-complete](/img/kb/vscode-schema-autocomplete-example.png)
+
+## Adding the Semgrep rule schema in VS Code
+
+Adding the Semgrep rule schema in VS Code requires two steps:
+
+1. Install the YAML Language Support extension by Red Hat
+2. Associate the Semgrep rule schema
+
+### Install the YAML Language Support extension by Red Hat
+
+You can install the extension either directly in VS Code or by going to the Visual Studio Marketplace and installing it from there. In VS Code, typically just searching `yaml` will yield the correction extension as the top result. However, please verify you are installing the correct extension by ensuring it is the same as the one [linked here](https://marketplace.visualstudio.com/items?itemName=redhat.vscode-yaml).
```

Comment:
I have some nitpicks about the language here, but more importantly I don't think it's a good idea for us to say "just search yaml" since that might easily change. What should they be looking for after searching YAML to confirm?

Reply:
The extension is actually just named "YAML", but I can be a bit more specific. Perhaps something like this:

```suggestion
You can install the extension either directly in VS Code or by going to the Visual Studio Marketplace and installing it from there. In VS Code, typically just searching `yaml` will yield the correction extension as the top result. The extension is named "YAML" and is authored by "Red Hat". However, please also verify you are installing the correct extension by ensuring it is the same as the one [linked here](https://marketplace.visualstudio.com/items?itemName=redhat.vscode-yaml).
```